lib/grinder_sdk.dart

```diff
@@ -146,6 +191,103 @@ class Pub {
   }
 }
 
+/**
+ * Utility tasks for invoking dart2js.
+ */
+class Dart2js {
+  /**
+   * Invoke a dart2js compile with the given [sourceFile] as input.
+   */
+  static void compile(File sourceFile,
+      {Directory outDir, bool minify: false, bool csp: false}) {
+    if (outDir == null) outDir = sourceFile.parent;
+    File outFile = joinFile(outDir, ["${fileName(sourceFile)}.js"]);
+
+    if (!outDir.existsSync()) outDir.createSync(recursive: true);
+
+    List args = [];
+    if (minify) args.add('--minify');
+    if (csp) args.add('--csp');
+    args.add('-o${outFile.path}');
+    args.add(sourceFile.path);
+
+    run_lib.run(_sdkBin('dart2js'), arguments: args);
+  }
+
+  /**
+   * Invoke a dart2js compile with the given [sourceFile] as input.
+   */
+  static Future compileAsync(File sourceFile,
+      {Directory outDir, bool minify: false, bool csp: false}) {
+    if (outDir == null) outDir = sourceFile.parent;
+    File outFile = joinFile(outDir, ["${fileName(sourceFile)}.js"]);
+
+    if (!outDir.existsSync()) outDir.createSync(recursive: true);
+
+    List args = [];
+    if (minify) args.add('--minify');
+    if (csp) args.add('--csp');
+    args.add('-o${outFile.path}');
+    args.add(sourceFile.path);
+
+    return run_lib.runAsync(_sdkBin('dart2js'), arguments: args)
+        .then((_) => null);
+  }
+
+  static String version({bool quiet: false}) =>
+      _AppVersion.parse(_run('--version', quiet: quiet)).version;
+
+  static String _run(String command, {bool quiet: false}) =>
+      run_lib.run(_sdkBin('dart2js'), quiet: quiet, arguments: [command]);
+}
+
+/**
+ * Utility tasks for invoking the analyzer.
+ */
+class Analyzer {
+  /// Analyze a single [File] or path ([String]).
+  static void analyze(fileOrPath,
+      {Directory packageRoot, bool fatalWarnings: false}) {
+    analyzeFiles([fileOrPath], packageRoot: packageRoot,
+        fatalWarnings: fatalWarnings);
+  }
+
+  /// Analyze one or more [File]s or paths ([String]).
+  static void analyzeFiles(List files,
+      {Directory packageRoot, bool fatalWarnings: false}) {
+    List args = [];
+    if (packageRoot != null) args.add('--package-root=${packageRoot.path}');
+    if (fatalWarnings) args.add('--fatal-warnings');
+    args.addAll(files.map((f) => f is File ? f.path : f));
+
+    run_lib.run(_sdkBin('dartanalyzer'), arguments: args);
+  }
+
+  static String version({bool quiet: false}) => _AppVersion.parse(run_lib.run(
+      _sdkBin('dartanalyzer'), quiet: quiet, arguments: ['--version'])).version;
+}
+
+/// Utility class for invoking `dartfmt` from the SDK.
+class DartFmt {
+  /// Run the `dartfmt` command with the `--overwrite` option. Format any files
+  /// in place.
+  static void format(fileOrPath) {
```

Comment:
I think we'd be better off moving everything over to constructors and instance methods instead of classes as containers for static methods.  That would allow them to be:

* configured via constructors
* passed around as instances
* mocked out for testing

So:

```dart
class Pub {
  Pub();
  get(...);
  //...
}

final Pub pub = new Pub();

class Formatter {
  Formatter.sdk(...);
  Formatter([VersionConstraint constraint]);
  //...
}
```

```dart
pub.global.run(...);
sdk.formatter.format(...); // As suggested in another review comment.
new Formatter().format(...); // Uses pub.
```